ONSdigital/rm-sample-service
# Enrichment of the sample summary

# What and why?
Step one of the new sample - collection exercise design. This recreates the validation step from collection exercise and does it in the sample service instead. However instead of calling it validation, it's called enrichment as that is what it's actually doing.

For each sample in the sample summary it should:
1. Find and store the party id
2. Find and store the correct collection instrument id
3. Determine if there are any active enrolments

# How to test?
Set up a collection exercise and load a sample but do not configure it to go live. Then send a HTTP request to the temporary end point below to trigger the enrichment process. Check the sample units in the database should and they should have the above values set.

```
http://localhost:8125/enrich/survey/02b9c366-7397-42f7-942a-76dc5876d86d/collection-exercise/941d94df-cf25-4ae1-bd1e-83ba6d6362c8/samplesummary/b1e33da4-923e-4f8d-b303-4aacc65539ec
```

# Trello
https://trello.com/c/E5yS2sIL/438-s37-ce-sample-validation-phase

Note this PR is set against a new sample-v2 branch.
